### PR TITLE
[SDL-0249] update RAI for multiple apps, improvement, fix typos

### PR DESCRIPTION
**Pull request by VjKlepikov**
### Summary
[Summary of PR changes]
- add test scripts:  022_1_Suspending_of_RAI_request_processing_in_case_cash_file_is_undefined.lua/022_1_Suspending_of_RAI_request_processing_in_case_cash_file_is_undefined.lua
Check that SDL suspend of RAI request processing from mobile app until all HMI Capabilities (VR/TTS/RC/UI/Buttons.GetCapabilities/,VR/TTS/UI.GetSupportedLanguages/GetLanguage, VehicleInfo.GetVehicleType) are received from HMI in case HMI capabilities cache file is undefined in smartDeviceLink.ini file
- update comon.registerAppsSuspend
- add preconditions to define HMICapabilitiesCacheFile
- move removedRaduantDisplayCapParameters, expCapRaiResponse to common module
- fix typos

### CODE REVIEW

File: test_scripts/Capabilities/PersistingHMICapabilities/common.lua [hunk 1]
```diff
@@ -556,19 +586,43 @@ function m.registerAppsSuspend( pCapResponse, pHMIParams )
...
-  :Do(function(exp, data)
-      if exp.occurences == 1 then
-        actions.app.setHMIId(data.params.application.appID, appSessionId1)
-      elseif exp.occurences == 2 then
-        actions.app.setHMIId(data.params.application.appID, appSessionId2)
-      else
-        actions.app.setHMIId(data.params.application.appID, appSessionId3)
-      end
-    end)
+  local isAppSessionId = {
```
Comment (aderiabin) [inline test_scripts/Capabilities/PersistingHMICapabilities/common.lua:589]: @VjKlepikov The name of the variable is not reflect its purpose. I would propose to rename it
```suggestion
  local isAppRegisteredReceived = {
```

File: test_scripts/Capabilities/PersistingHMICapabilities/common.lua [hunk 2]
```diff
@@ -556,19 +586,43 @@ function m.registerAppsSuspend( pCapResponse, pHMIParams )
...
+      validationBCOnAppRegistered(appSessionId1)
+    elseif appName == actions.app.getParams(appSessionId2).appName then
+      validationBCOnAppRegistered(appSessionId2)
+    elseif appName == actions.app.getParams(appSessionId3).appName then
+      validationBCOnAppRegistered(appSessionId3)
+    else
+      actions.run.fail("SDL sends unexpected BasicCommunication.OnAppRegistered notification for app" .. appName)
+    end
+    return true
+  end)
```
Comment (aderiabin) [inline test_scripts/Capabilities/PersistingHMICapabilities/common.lua:614]: @VjKlepikov I would propose to simplify this validation:
```suggestion
local isAppRegisteredReceived = {
  [actions.app.getParams(appSessionId1).appName] = false,
  [actions.app.getParams(appSessionId2).appName] = false,
  [actions.app.getParams(appSessionId3).appName] = false
}

local function validateBCOnAppRegistered(pAppName)
  if isAppRegisteredReceived[pAppName] == false then
    isAppRegisteredReceived[pAppName] = true
    return true
  elseif isAppRegisteredReceived[pAppName] == nil then
    return false, "SDL sends unexpected BC.OnAppRegistered notification for app with name " .. pAppName
  else
    return false, "SDL sends BC.OnAppRegistered notification twice for app with name" .. pAppName
  end
end

actions.hmi.getConnection():ExpectNotification("BasicCommunication.OnAppRegistered")
:ValidIf(function(_,data)
  return validateBCOnAppRegistered(data.params.application.appName)
end)
```
=>
```suggestion
local isAppRegisteredReceived = {
  [actions.app.getParams(appSessionId1).appName] = false,
  [actions.app.getParams(appSessionId2).appName] = false,
  [actions.app.getParams(appSessionId3).appName] = false
}

actions.hmi.getConnection():ExpectNotification("BasicCommunication.OnAppRegistered")
:ValidIf(function(_, data)
  local appName = data.params.application.appName
  if isAppRegisteredReceived[appName] == false then
    isAppRegisteredReceived[appName] = true
    return true
  elseif isAppRegisteredReceived[appName] == nil then
    return false, "SDL sends unexpected BasicCommunication.OnAppRegistered notification for app with name " .. appName
  else
    return false, "SDL sends BasicCommunication.OnAppRegistered notification twice for app with name" .. appName
  end
end)
```

File: test_sets/persisting_hmi_capabilities.txt [hunk 3]
```diff
@@ -22,6 +22,7 @@
 ./test_scripts/Capabilities/PersistingHMICapabilities/020_RAI_use_use_cash_capability.lua
 ./test_scripts/Capabilities/PersistingHMICapabilities/021_Suspending_of_RAI_request_processing_with_another_ccpu_version.lua
 ./test_scripts/Capabilities/PersistingHMICapabilities/022_Suspending_of_RAI_request_processing_in_case_cash_file_does_not_exist.lua
+./test_scripts/Capabilities/PersistingHMICapabilities/022_1_Suspending_of_RAI_request_processing_in_case_cash_file_is_undefined.lua
```
Comment (aderiabin) [inline test_sets/persisting_hmi_capabilities.txt:25]: @VjKlepikov Please re-enumerate test scripts in a row.

File: test_scripts/Capabilities/PersistingHMICapabilities/008_HMI_sends_GetSystemInfo_with_another_ccpu_version.lua [hunk 4]
```diff
@@ -1,15 +1,16 @@
...
+--  SDL does requested all capabilities from HMI in case ccpu_version do not match
 --
 -- Preconditions:
--- 1. hmi_capabilities_cache.json file doesn't exist on file system
--- 2. SDL and HMI are started
--- 3. HMI sends all HMI capabilities (VR/TTS/RC/UI etc) to SDL
--- 4. SDL stored capability to "hmi_capabilities_cache.json" file in AppStorageFolder
--- 5. Ignition OFF/ON cycle performed
+-- 1  Value of HMICapabilitiesCacheFile parameter is defined (hmi_capabilities_cache.json) in smartDeviceLink.ini file
+-- 2. HMI capabilities cache file (hmi_capabilities_cache.json) file doesn't exist on file system
```
[the remark this answers is not in the document]
Comment (VjKlepikov) [reply test_scripts/Capabilities/PersistingHMICapabilities/008_HMI_sends_GetSystemInfo_with_another_ccpu_version.lua]: Fixed (see 1b35cfd)

File: test_scripts/Capabilities/PersistingHMICapabilities/010_1_HMI_sends_GetSystemInfo_response_with_another_language.lua [hunk 5]
```diff
@@ -5,11 +5,12 @@
 --  in case HMI sends BC.GetSystemInfo response with another language/wersCountryCode
 --
 -- Preconditions:
--- 1. HMI sends GetSystemInfo with ccpu_version = "ccpu_version_1", language = "EN-US", wersCountryCode = "wersCountryCode_1" to SDL
--- 2. HMI sends all HMI capabilities (VR/TTS/RC/UI etc)
--- 3. SDL persists capabilities to HMI capabilities cache file ("hmi_capabilities_cache.json") in AppStorageFolder
--- 4. Ignition OFF/ON cycle performed
--- 5. SDL is started and send GetSystemInfo request
+-- 1  Value of HMICapabilitiesCacheFile parameter is defined (hmi_capabilities_cache.json) in smartDeviceLink.ini file
+-- 2. HMI sends GetSystemInfo with ccpu_version = "ccpu_version_1", language = "EN-US", wersCountryCode = "wersCountryCode_1" to SDL
```
[the remark this answers is not in the document]
Comment (VjKlepikov) [reply test_scripts/Capabilities/PersistingHMICapabilities/010_1_HMI_sends_GetSystemInfo_response_with_another_language.lua]: Fixed (see acf955)

File: test_scripts/Capabilities/PersistingHMICapabilities/011_HMI_sends_GetSystemInfo_response_with_incorrect_data.lua [hunk 6]
```diff
@@ -1,15 +1,16 @@
...
--- 4. SDL stored capability to "hmi_capabilities_cache.json" file in AppStorageFolder
--- 5. Ignition OFF/ON cycle performed
+-- 1  Value of HMICapabilitiesCacheFile parameter is defined (hmi_capabilities_cache.json) in smartDeviceLink.ini file
+-- 2. HMI capabilities cache file (hmi_capabilities_cache.json) file doesn't exist on file system
+-- 3. SDL and HMI are started
+-- 4. HMI sends all HMI capabilities
+-- 5. SDL stored capabilities to "hmi_capabilities_cache.json" file in AppStorageFolder
+-- 6. Ignition OFF/ON cycle performed
 -- Sequence:
 -- 5. HMI sends "BasicCommunication.GetSystemInfo" response without mandatory parameter "ccpu_version"/ invalid parameter type
```
[the remark this answers is not in the document]
Comment (VjKlepikov) [reply test_scripts/Capabilities/PersistingHMICapabilities/011_HMI_sends_GetSystemInfo_response_with_incorrect_data.lua]: Fixed (see acf955)

File: test_scripts/Capabilities/PersistingHMICapabilities/common.lua (hunk 2 above)
Comment (VjKlepikov) [reply test_scripts/Capabilities/PersistingHMICapabilities/common.lua]: Fixed (see 3ab319)

File: test_scripts/Capabilities/PersistingHMICapabilities/common.lua [hunk 7]
```diff
@@ -66,6 +65,37 @@ function m.noResponseGetHMIParams()
...
+end
+
+function m.removedRaduantDisplayCapParameters(pDisplayCapabilities)
+  local displayCapabilities = pDisplayCapabilities
+  displayCapabilities.imageCapabilities = nil  -- no Mobile_API.xml
+  displayCapabilities.menuLayoutsAvailable = nil --since 6.0
+  return displayCapabilities
+end
+
+function m.expCapRaiResponse()
```
[the remark this answers is not in the document]
Comment (VjKlepikov) [reply test_scripts/Capabilities/PersistingHMICapabilities/common.lua]: Fixed (see ad6707)

File: test_scripts/Capabilities/PersistingHMICapabilities/common.lua (hunk 1 above)
Comment (VjKlepikov) [reply test_scripts/Capabilities/PersistingHMICapabilities/common.lua]: Fixed (see 3ab319)

File: test_scripts/Capabilities/PersistingHMICapabilities/common.lua [hunk 8]
```diff
@@ -556,19 +586,43 @@ function m.registerAppsSuspend( pCapResponse, pHMIParams )
...
-      else
-        actions.app.setHMIId(data.params.application.appID, appSessionId3)
-      end
-    end)
+  local isAppSessionId = {
+    [1] = false,
+    [2] = false,
+    [3] = false
+  }
+  local function validationBCOnAppRegistered(pAppId)
```
[the remark this answers is not in the document]
Comment (VjKlepikov) [reply test_scripts/Capabilities/PersistingHMICapabilities/common.lua]: Fixed (see 3ab319)

File: test_sets/persisting_hmi_capabilities.txt (hunk 3 above)
Comment (VjKlepikov) [reply test_sets/persisting_hmi_capabilities.txt]: Fixed (see [4f166](https://github.com/LuxoftSDL/sdl_atf_test_scripts/pull/12/commits/4f1669b77acdb41ae2f36e9c2f21e0d6d1ea10ff))

**Pull request merged**